### PR TITLE
Add Pytorch buffers node

### PR DESCRIPTION
**Pull request by lapid92**

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 1]
```diff
@@ -123,10 +145,16 @@ def nodes_builder(model: GraphModule,
             weights.update(named_buffer_weights)
 
         if node.op == GET_ATTR:
-            named_parameters_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
-                                        model.named_parameters() if node.target == name}
-            named_buffer_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
-                                    model.named_buffers() if node.target == name}
+            if node_type == BufferHolder:
+                named_parameters_weights = {CONSTANT: parameter for name, parameter in
```
Comment (haihabi) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:149]: Make this a function the code repart itself

File: model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py [hunk 2]
```diff
@@ -160,7 +162,10 @@ def _quantize_node_activations(self,
     def _add_modules(self):
         for n in self.node_sort:
             if not isinstance(n, FunctionalNode):
-                self.add_module(n.name, node_builder(n))
+                    if n.type == BufferHolder:
+                        self.register_buffer(n.name, n.weights['constant'])
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py:166]: Use constants please

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 3]
```diff
@@ -52,6 +52,24 @@ def forward(self):
...
+class BufferHolder(torch.nn.Module):
+    """
+    Class for saving buffers in graph inference.
+    """
+    def __init__(self, const_size):
+        super(BufferHolder, self).__init__()
+        if torch.cuda.is_available():
+            setattr(self, CONSTANT, torch.FloatTensor(size=const_size).to(torch.device('cuda')))
+        else:
+            setattr(self, CONSTANT, torch.FloatTensor(size=const_size))
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:64]: don't need this "if". please use "get_working_device" from utils

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 4]
```diff
@@ -104,11 +122,15 @@ def nodes_builder(model: GraphModule,
...
                 raise Exception(f'Call method of type \'{node.target}\' is currently not supported.')
         elif node.op == GET_ATTR:
-            node_type = ConstantHolder
-            node_has_activation = False
-            common.Logger.warning(
-                'Pytorch model has a parameter or constant Tensor value. This can cause unexpected behaviour when '
-                'converting the model.')
+            if node.meta[TYPE] == torch.Tensor:
+                node_type = BufferHolder
+                node_has_activation = False
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:127]: add a warning similar to ConstantHolder

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 5]
```diff
@@ -157,6 +185,19 @@ def nodes_builder(model: GraphModule,
...
+            if not isinstance(v, torch.fx.node.Node):
+                framework_attr_filtered[k] = v
+        framework_attr = framework_attr_filtered
+
+        # filter Nodes from node kwargs
+        node_kwargs = {}
+        for k, v in node.kwargs.items():
+            if not isinstance(v, torch.fx.node.Node):
+                node_kwargs[k] = v
+
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:200]: maybe add a comment to explain this code. Say that we replace these attributes with nx graph nodes

File: model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py [hunk 6]
```diff
@@ -84,6 +84,8 @@ def _run_operation(n: BaseNode,
     functional_kwargs = n.op_call_kwargs if isinstance(n, FunctionalNode) else {}
     if isinstance(n, FunctionalNode) and n.inputs_as_list:
         out_tensors_of_n = op_func(input_tensors, *op_call_args, **functional_kwargs)
+    elif n.type == BufferHolder:
+        out_tensors_of_n = n.weights['constant']
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/back2framework/pytorch_model_builder.py:88]: Why do you need this? can't you implement it similar to ConstantHolder? add a "forward" method to BufferHolder

File: model_compression_toolkit/core/pytorch/reader/graph_builders.py [hunk 7]
```diff
@@ -52,6 +52,24 @@ def forward(self):
         return getattr(self, CONSTANT)
 
 
+class BufferHolder(torch.nn.Module):
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/reader/graph_builders.py:55]: Maybe we need to move these classes to a different file? "node_holders" or something of the sort? @haihabi what do you think?

Comment (haihabi) [reply model_compression_toolkit/core/pytorch/reader/graph_builders.py:55]: Agree

**Pull request closed without merging**